src/Compilers/CSharp/Test/Symbol/SymbolDisplay/SymbolDisplayTests.cs

```diff
@@ -11,13 +11,62 @@
 using Microsoft.CodeAnalysis.CSharp.Test.Utilities;
 using Microsoft.CodeAnalysis.Test.Utilities;
 using Roslyn.Test.Utilities;
+using Roslyn.Utilities;
 using Xunit;
 
 namespace Microsoft.CodeAnalysis.CSharp.UnitTests
 {
     public class SymbolDisplayTests : CSharpTestBase
     {
         [Fact]
+        [WorkItem(22507, "https://github.com/dotnet/roslyn/issues/22507")]
+        public void Repro22507()
+        {
+            var text = @"
+using System;
+
+[Flags]
+public enum SomeTypes : long
+{
+    Default = 0,
+    TypeA = 2,
+    TypeB = 4,
+    TypeC = 1024,
+    TypeD = 1048576,
+    TypeE = 2097152,
+    TypeF = 33554432,
+    TypeG = 67108864,
+    TypeH = 134217728,
+    TypeI = 1073741824,
+    TypeJ = 2147483648,
+    TypeK = 4294967296,
+    TypeL = 1099511627776,
+    TypeM = 2199023255552,
+    TypeN = 4398046511104,
+    TypeO = 1125899906842624,
+    TypeP = 2251799813685248,
+    TypeQ = 4503599627370496,
+    TypeR = 9007199254740992,
+    TypeS = 18014398509481984,
+    TypeT = 2305843009213693952
+}
```

Comment:
Is there a simpler repro, with just two values perhaps?

Reply:
I'm certain there is, but isn't it better to use the full repro as a regression test?

Reply:
I've added a unit test to directly demonstrate the enum comparer problem since the customer repro is very close to a minimal repro to actually produce an exception.